packages/core/lib/internals/Fiber.ts

```diff
@@ -263,19 +263,14 @@ export class FiberContext<A, E> extends Fiber<A, E> implements ICancellable {
             data = j.i0(env)
             break
           case Tag.Async:
-            const id = this.cancellationList.push(
-              j.i0(
-                (val) => {
-                  this.cancellationList.remove(id)
-                  this.stackA.push(QIO.resolve(val).asInstruction)
-                  this.unsafeEvaluate()
-                },
-                (err) => {
-                  this.cancellationList.remove(id)
-                  this.stackA.push(QIO.reject(err).asInstruction)
-                  this.unsafeEvaluate()
+            this.asyncOp = this.cancellationList.push(
```

Comment:
using `this.asyncOp` is going to be slower while setting.  Revert back to using `id`